src/utils.py

```diff
@@ -159,12 +160,22 @@ def convert_version_to_proper_semantic(version):
     1.5.2-RELEASE for semantic version to work'"""
     if version in ('', '-1', None):
         return '0.0.0'
+    vsplit_components = version.split('.')
+    len_vsplit = len(vsplit_components)
+    if len_vsplit == 1:
+        version = version + ".0.0"
+    elif len_vsplit == 2:
+        version = version + ".0"
```

Comment:
Have you considered using [semantic_version.Version.coerce()](https://python-semanticversion.readthedocs.io/en/latest/#coercing-an-arbitrary-version-string) ?

Reply:
@jpopelka No, I did not know about that, just checked, it will definitely solve some of the problems for us. For a quick fix I will update the PR based on the coerce function.
But as pointed in the issue, we will need a deeper discussion and better solution to handle it across f8a for various ecosystems.